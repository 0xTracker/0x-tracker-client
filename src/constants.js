const API_ENDPOINT = process.env.REACT_APP_API_ENDPOINT;

const API_CALL_TIMEOUT = process.env.REACT_APP_API_CALL_TIMEOUT || 30000;

const BREAKPOINTS = {
  xs: 575,
  sm: 767, // eslint-disable-line sort-keys
  md: 991, // eslint-disable-line sort-keys
  lg: 1199, // eslint-disable-line sort-keys
  xl: Infinity,
};

const TIME_PERIOD = {
  ALL: 'all',
  DAY: 'day',
  MONTH: 'month',
  WEEK: 'week',
  YEAR: 'year',
};

const DATE_FORMAT = {
  COMPACT: 'compact',
  FULL: 'fill',
  RELATIVE: 'relative',
  STANDARD: 'standard',
};

const GENESIS_DATE = new Date('2017-08-15T00:00:00Z');

const URL = {
  FILL: '/fills/:id',
  FILLS: '/fills',
  HOME: '/',
  NETWORK_INSIGHTS: '/network-insights',
  NEWS: '/news-and-updates',
  RELAYER: '/relayers/:slug',
  RELAYERS: '/relayers',
  SEARCH: '/search',
  TOKEN: '/tokens/:address',
  TOKENS: '/tokens',
  TRADER: '/traders/:address',
  TRADERS: '/traders',
};

const ETH_TOKEN = {
  symbol: 'ETH',
};

const ZRX_TOKEN = {
  address: '0xe41d2489571d322189246dafa5ebde1f4699f498',
  symbol: 'ZRX',
};

export {
  API_CALL_TIMEOUT,
  API_ENDPOINT,
  BREAKPOINTS,
  DATE_FORMAT,
  GENESIS_DATE,
  TIME_PERIOD,
  URL,
  ETH_TOKEN,
  ZRX_TOKEN,
};
