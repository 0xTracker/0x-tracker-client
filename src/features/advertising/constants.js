const FATHOM_ADVERT = {
  DESCRIPTION:
    'Privacy focused alternative to Google Analytics. Free for 7 days, then $10 off your first bill ',
  ICON_URL: 'https://resources.0xtracker.com/logos/fathom.png',
  TITLE: 'Fathom Analytics',
  URL: 'https://usefathom.com/ref/TFTVTT',
};

const BRAVE_ADVERT = {
  DESCRIPTION:
    'Secure your browser and earn crypto rewards just for browsing the web',
  ICON_URL:
    'https://brave.com/wp-content/uploads/2018/02/cropped-brave_appicon_release-180x180.png',
  TITLE: 'Brave Browser',
  URL: 'https://brave.com/xtr583',
};

export { BRAVE_ADVERT, FATHOM_ADVERT };
