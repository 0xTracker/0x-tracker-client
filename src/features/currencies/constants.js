const BASE_CURRENCY = 'USD';

const CURRENCIES = [
  { name: '$ Australian Dollar', symbol: 'AUD' },
  { name: '₿ Bitcoin', symbol: 'BTC' },
  { name: '£ British Sterling', symbol: 'GBP' },
  { name: '¥ Chinese Yuan', symbol: 'CNY' },
  { name: 'Ξ Ether', symbol: 'ETH' },
  { name: '€ Euro', symbol: 'EUR' },
  { name: '¥ Japanese Yen', symbol: 'JPY' },
  { name: '￦ Korean Won', symbol: 'KRW' },
  { name: '$ US Dollar', symbol: 'USD' },
];

export { BASE_CURRENCY, CURRENCIES };
