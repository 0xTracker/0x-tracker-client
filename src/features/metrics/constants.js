// eslint-disable-next-line import/prefer-default-export
export const METRIC_TYPE = {
  TOKEN_VOLUME: 'token-volume',
};
