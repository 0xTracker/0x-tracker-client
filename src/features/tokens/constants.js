const TOKEN_TYPE = {
  ERC20: 'erc-20',
  ERC721: 'erc-721',
};

export { TOKEN_TYPE };
