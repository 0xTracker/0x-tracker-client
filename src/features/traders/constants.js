const TRADER_TYPE = {
  MAKER: 'maker',
  TAKER: 'taker',
};

export { TRADER_TYPE };
