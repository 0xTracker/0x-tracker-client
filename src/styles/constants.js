const breakpoints = {
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200,
};

const colors = {
  blueMarguerite: '#867ac6',
  bonJour: '#dfdbdb',
  bostonBlue: '#3992CA',
  charade: '#2D2D38',
  cinder: '#151521',
  cranberry: '#E24F8B',
  gallery: '#ebebeb',
  goldenSand: '#F0DB79',
  halfBaked: '#8DC6C4',
  mineShaft: '#222',
  moodyBlue: '#7a68d2',
  persianBlue: '#1e42db',
  scampi: '#695f9c',
  shark: '#27272b',
  silver: '#c0c0c0',
  tuna: '#363a40',
  white: '#fff',
  wildSand: '#f4f4f4',
};

export { breakpoints, colors };
