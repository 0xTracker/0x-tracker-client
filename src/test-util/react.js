import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
