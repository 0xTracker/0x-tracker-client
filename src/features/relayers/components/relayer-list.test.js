import React from 'react';

import { renderWithAppContext } from '../../../../test-utils/react';
import { TIME_PERIOD } from '../../../constants';
import RelayerList from './relayer-list';

const basicProps = {
  relayers: [
    {
      id: 'radarRelay',
      name: 'Radar Relay',
      slug: 'radar-relay',
      stats: {
        trades: 512,
        volume: { USD: 100000 },
      },
      url: 'https://radarrelay.com',
    },
    {
      id: 'ddex',
      name: 'DDEX',
      slug: 'ddex',
      stats: {
        trades: 256,
        volume: { USD: 150456.56 },
      },
      url: 'https://ddex.io',
    },
    {
      id: 'paradex',
      name: 'Paradex',
      slug: 'paradex',
      stats: {
        trades: 345,
        volume: { USD: 124000 },
      },
      url: 'https://paradex.io',
    },
    {
      id: 'sadRelayer',
      name: 'Sad Relayer',
      slug: 'sad-relayer',
      stats: {
        trades: 0,
        volume: { USD: 0 },
      },
      url: 'https://google.com',
    },
  ],
  timePeriod: TIME_PERIOD.DAY,
};

it('renders with basic props', () => {
  const { container } = renderWithAppContext(<RelayerList {...basicProps} />);

  expect(container.firstChild).toMatchSnapshot();
});

it('renders for one month time period', () => {
  const { getByText } = renderWithAppContext(
    <RelayerList {...basicProps} timePeriod={TIME_PERIOD.MONTH} />,
  );

  getByText('Volume ( 1M )');
});
