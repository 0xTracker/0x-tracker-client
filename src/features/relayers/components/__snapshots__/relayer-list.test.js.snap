// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with basic props 1`] = `
.c3 {
  color: #3f4ac3;
  cursor: pointer;
}

.c3:hover {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  color: #3f4ac3;
  cursor: pointer;
}

.c0:hover {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  display: inline-block;
  vertical-align: -.125em;
  overflow: hidden;
}

.c2 {
  color: #6C6B80;
  font-size: 0.9rem;
}

<table
  class="table table-responsive"
>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th
        colspan="2"
      >
        Relayer
      </th>
      <th
        class="text-right"
      >
        Trades (
        24H
        )
      </th>
      <th
        class="text-right"
      >
        Volume (
        24H
        )
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        class="align-middle"
      >
        1
      </td>
      <td
        class="align-middle"
      >
        <a
          class="c0"
          href="/relayers/ddex"
        >
          <svg
            aria-hidden="true"
            class="c1"
            color="#dcdde4"
            fill="currentColor"
            focusable="false"
            height="40"
            viewBox="0 0 448 512"
            width="40"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
            />
          </svg>
        </a>
      </td>
      <td
        class="align-middle"
        width="99%"
      >
        <a
          class="c0"
          href="/relayers/ddex"
        >
          DDEX
        </a>
        <br />
        <a
          _css="#6C6B80"
          class="c2 c3"
          href="https://ddex.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://ddex.io
        </a>
      </td>
      <td
        class="align-middle text-right"
      >
        256
      </td>
      <td
        class="align-middle text-right"
      >
        <span>
          $150,456.56
        </span>
      </td>
    </tr>
    <tr>
      <td
        class="align-middle"
      >
        2
      </td>
      <td
        class="align-middle"
      >
        <a
          class="c0"
          href="/relayers/paradex"
        >
          <svg
            aria-hidden="true"
            class="c1"
            color="#dcdde4"
            fill="currentColor"
            focusable="false"
            height="40"
            viewBox="0 0 448 512"
            width="40"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
            />
          </svg>
        </a>
      </td>
      <td
        class="align-middle"
        width="99%"
      >
        <a
          class="c0"
          href="/relayers/paradex"
        >
          Paradex
        </a>
        <br />
        <a
          _css="#6C6B80"
          class="c2 c3"
          href="https://paradex.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://paradex.io
        </a>
      </td>
      <td
        class="align-middle text-right"
      >
        345
      </td>
      <td
        class="align-middle text-right"
      >
        <span>
          $124,000.00
        </span>
      </td>
    </tr>
    <tr>
      <td
        class="align-middle"
      >
        3
      </td>
      <td
        class="align-middle"
      >
        <a
          class="c0"
          href="/relayers/radar-relay"
        >
          <svg
            aria-hidden="true"
            class="c1"
            color="#dcdde4"
            fill="currentColor"
            focusable="false"
            height="40"
            viewBox="0 0 448 512"
            width="40"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
            />
          </svg>
        </a>
      </td>
      <td
        class="align-middle"
        width="99%"
      >
        <a
          class="c0"
          href="/relayers/radar-relay"
        >
          Radar Relay
        </a>
        <br />
        <a
          _css="#6C6B80"
          class="c2 c3"
          href="https://radarrelay.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://radarrelay.com
        </a>
      </td>
      <td
        class="align-middle text-right"
      >
        512
      </td>
      <td
        class="align-middle text-right"
      >
        <span>
          $100,000.00
        </span>
      </td>
    </tr>
    <tr
      class="faded"
    >
      <td
        class="align-middle"
      >
        4
      </td>
      <td
        class="align-middle"
      >
        <a
          class="c0"
          href="/relayers/sad-relayer"
        >
          <svg
            aria-hidden="true"
            class="c1"
            color="#dcdde4"
            fill="currentColor"
            focusable="false"
            height="40"
            viewBox="0 0 448 512"
            width="40"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
              fill="currentColor"
            />
          </svg>
        </a>
      </td>
      <td
        class="align-middle"
        width="99%"
      >
        <a
          class="c0"
          href="/relayers/sad-relayer"
        >
          Sad Relayer
        </a>
        <br />
        <a
          _css="#6C6B80"
          class="c2 c3"
          href="https://google.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          https://google.com
        </a>
      </td>
      <td
        class="align-middle text-right"
      >
        -
      </td>
      <td
        class="align-middle text-right"
      >
        -
      </td>
    </tr>
  </tbody>
</table>
`;
